editor/src/clj/editor/defold_project.clj

```diff
@@ -721,8 +721,14 @@
                                     ;; loaded.
                                     (or (new-node-id+resource-pairs-by-proj-path required-dependency-proj-path) ; If it was defunloaded, it might not have made it into the principal set, but now that it is unsafely referenced we do need it.
                                         (let [old-node-id (old-node-ids-by-proj-path required-dependency-proj-path)]
-                                          (if (nil? old-node-id)
+                                          (cond
+                                            (nil? old-node-id)
                                             nil ; The proj-path refers to a resource that does not exist. We should exclude it here, and it will eventually become a defective node in the graph as we process the :load-fn of the referencing resource.
+
+                                            (nil? old-node-id->old-node-state)
+                                            (throw (IllegalArgumentException. "Required old-node-id->old-node-state function was not provided."))
```

Comment:
Should we, perhaps, assert that `old-node-id->old-node-state` is provided at the beginning of the function?

Reply:
It is only necessary to provide it if you're reading nodes into an existing graph. It is not provided during the initial project load, only when `read-nodes` is called in response to a resource-sync, so it is not needed in most cases.